feature-libs/my-account/organization/components/user/form/user-form.service.ts

```diff
@@ -36,12 +30,28 @@ export class UserFormService {
     );
     form.setControl('roles', new FormArray([]));
     form.setControl('isAssignedToApprovers', new FormControl(false));
-  }
 
-  protected pathRoles(form, model) {
-    const roles = form.get('roles') as FormArray;
-    model.roles?.forEach((role) => {
-      roles.push(new FormControl(role));
+    form.get('roles').valueChanges.subscribe((roles: string[]) => {
```

Comment:
user form -> "Add the user to approvers for the unit" checkbox should be visible only if approver role is checked. That mechanism seems as not working

Reply:
Yes, we moved to a disabled state instead. The whole feature seems buggy, and we need to pick it up separately; doesn't seem to work on my company branch as well.

Reply:
I'm going to add a bit more content to #8418 to help the dev understanding the issue